src/Compilers/VisualBasic/Portable/Scanner/ScannerBuffer.vb

```diff
@@ -95,6 +96,36 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return p
         End Function
 
+        Friend Function TryPeek(at As Integer, <Out()> ByRef ch As Char) As Boolean
```

Comment:
Sorry, I didn't make it clear that two suggestions about this function were related.
If we want this function to follow TryGetValue pattern, which is a pretty well-known pattern, then 
1. Second parameter should be Out parameter (decorated with Out attribute)
2. Second parameter should be always assigned before the function returns.

Unless #2 is met, applying Out attribute is a violation of a contract for out parameters.

It is fine if we don’t want to follow the TryGetValue pattern, but then we shouldn’t have the Out attribute on the parameter. We should also explicitly state contract of the method in the comment. In addition we should consider slightly changing the name of the method so that consumers were not confused by the name similarity to TryGetValue. 

This comment also applies to the next function.